src/segment-loader.js

```diff
@@ -975,6 +976,22 @@ export default class SegmentLoader extends videojs.EventTarget {
     // equal to the last appended mediaIndex
     if (this.mediaIndex !== null) {
       this.mediaIndex -= mediaSequenceDiff;
+
+      // mediaIndex is invalid, set it to null
+      if (this.mediaIndex < 0) {
+        this.mediaIndex = null;
+        this.partIndex = null;
+      } else {
+        const segment = this.playlist_.segments[this.mediaIndex];
+
+        // partIndex should remain the same for the same segment
+        // unless parts fell off of the playlist for this segment.
+        // In that case we need to reset partIndex and resync
+        if (this.partIndex && (!segment.parts || !segment.parts.length || !segment.parts[this.partIndex])) {
+          this.logger_(`part fell off on part ${this.partIndex}`);
+          this.resetLoader();
```

Comment:
Not sure of a better way to do this. If we appended part 4 of 8 but parts fell off the manifest, they will no longer be request-able, so we kind of have to reset partIndex and the buffer. Perhaps the solution is to never use the first sequence of parts in a manifest?